providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdGrpcInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+import javax.annotation.Nullable;
+
+/**
+ * FlagdGrpcInterceptor is an interceptor for grpc communication from java-sdk to flagd.
+ * <a href="https://github.com/open-telemetry/opentelemetry-java-docs">credits</a>
+ */
+final class FlagdGrpcInterceptor implements ClientInterceptor {
+    private static final TextMapSetter<Metadata> SETTER = new Setter();
+
+    private final OpenTelemetry openTelemetry;
+
+    FlagdGrpcInterceptor(final OpenTelemetry openTelemetry) {
```

Comment:
Maybe we can add `@NotNull` so it is easier to understand the contract of the class

Reply:
added the annotation :)